internal/backend/remote-state/s3/backend_state.go

```diff
@@ -29,25 +30,29 @@ func (b *Backend) Workspaces() ([]string, error) {
 		prefix = b.workspaceKeyPrefix + "/"
 	}
 
-	params := &s3.ListObjectsInput{
+	params := &s3.ListObjectsV2Input{
 		Bucket:  &b.bucketName,
 		Prefix:  aws.String(prefix),
-		MaxKeys: aws.Int64(maxKeys),
+		MaxKeys: maxKeys,
 	}
 
 	wss := []string{backend.DefaultStateName}
-	err := b.s3Client.ListObjectsPages(params, func(page *s3.ListObjectsOutput, lastPage bool) bool {
+	paginator := s3.NewListObjectsV2Paginator(b.s3Client, params)
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(context.TODO())
+		if err != nil {
+			var noSuchBucket *types.NoSuchBucket
+			if errors.As(err, &noSuchBucket) {
+				return nil, fmt.Errorf(errS3NoSuchBucket, err)
+			}
+		}
```

Comment:
What should we do here if an unexpected error is occurred? Previous implementation does not have any error handling for this too, and that causes a panic.